vortex-array/src/array/constant/compute/mod.rs

```diff
@@ -0,0 +1,146 @@
+mod boolean;
+
+use std::cmp::Ordering;
+
+use vortex_error::VortexResult;
+use vortex_scalar::Scalar;
+
+use crate::array::constant::ConstantArray;
+use crate::array::ConstantEncoding;
+use crate::compute::unary::ScalarAtFn;
+use crate::compute::{
+    scalar_cmp, ArrayCompute, BinaryBooleanFn, ComputeVTable, FilterFn, FilterMask, MaybeCompareFn,
+    Operator, SearchResult, SearchSortedFn, SearchSortedSide, SliceFn, TakeFn, TakeOptions,
+};
+use crate::{ArrayData, ArrayLen, IntoArrayData};
+
+impl ArrayCompute for ConstantArray {
+    fn compare(&self, other: &ArrayData, operator: Operator) -> Option<VortexResult<ArrayData>> {
+        MaybeCompareFn::maybe_compare(self, other, operator)
+    }
+
+    fn search_sorted(&self) -> Option<&dyn SearchSortedFn> {
+        Some(self)
+    }
+}
+
+impl ComputeVTable for ConstantEncoding {
+    fn binary_boolean_fn(
+        &self,
+        lhs: &ArrayData,
+        rhs: &ArrayData,
+    ) -> Option<&dyn BinaryBooleanFn<ArrayData>> {
+        // We only need to deal with this if both sides are constant, otherwise other arrays
+        // will have handled the RHS being constant.
+        (lhs.is_constant() && rhs.is_constant()).then_some(self)
+    }
+
+    fn filter_fn(&self) -> Option<&dyn FilterFn<ArrayData>> {
+        Some(self)
+    }
+
+    fn scalar_at_fn(&self) -> Option<&dyn ScalarAtFn<ArrayData>> {
+        Some(self)
+    }
+
+    fn slice_fn(&self) -> Option<&dyn SliceFn<ArrayData>> {
+        Some(self)
+    }
+
+    fn take_fn(&self) -> Option<&dyn TakeFn<ArrayData>> {
+        Some(self)
+    }
+}
+
+impl ScalarAtFn<ConstantArray> for ConstantEncoding {
+    fn scalar_at(&self, array: &ConstantArray, _index: usize) -> VortexResult<Scalar> {
+        Ok(array.owned_scalar())
+    }
+}
+
+impl TakeFn<ConstantArray> for ConstantEncoding {
+    fn take(
+        &self,
+        array: &ConstantArray,
+        indices: &ArrayData,
+        _options: TakeOptions,
+    ) -> VortexResult<ArrayData> {
+        Ok(ConstantArray::new(array.owned_scalar(), indices.len()).into_array())
+    }
+}
+
+impl SliceFn<ConstantArray> for ConstantEncoding {
+    fn slice(&self, array: &ConstantArray, start: usize, stop: usize) -> VortexResult<ArrayData> {
+        Ok(ConstantArray::new(array.owned_scalar(), stop - start).into_array())
+    }
+}
+
+impl FilterFn<ConstantArray> for ConstantEncoding {
+    fn filter(&self, array: &ConstantArray, mask: FilterMask) -> VortexResult<ArrayData> {
+        Ok(ConstantArray::new(array.owned_scalar(), mask.true_count()).into_array())
+    }
+}
+
+impl SearchSortedFn for ConstantArray {
+    fn search_sorted(&self, value: &Scalar, side: SearchSortedSide) -> VortexResult<SearchResult> {
+        match self
+            .scalar_value()
+            .partial_cmp(value.value())
+            .unwrap_or(Ordering::Less)
```

Comment:
I thought we sorted null last? I'd expect `NULL cmp X` to be `Ordering::Greater`.

Reply:
While I may well agree, I'm just code shoveling! Can we FLUP?

Reply:
this was the case before, this is not about null comparison but where two types don't match, we should error here probably

Reply:
https://github.com/spiraldb/vortex/issues/1412